src/CoreBundle/Bridge/Chameleon/Module/NavigationTree/NavigationTree.php

```diff
@@ -0,0 +1,779 @@
+        }
+
+        $updatedNodes = [];
+
+        $entrySortField = $this->getSortFieldName();
+
+        $quotedTreeTable = $this->dbConnection->quoteIdentifier($this->treeTable);
+        $quotedParentNodeId = $this->dbConnection->quote($parentNodeId);
+        if (0 == $position) { // set every other node pos+1
+            $query = 'SELECT * FROM ' . $quotedTreeTable . ' WHERE parent_id = ' . $quotedParentNodeId;
+            $cmsTreeList = \TdbCmsTreeList::GetList($query);
+            while ($cmsTreeNode = $cmsTreeList->Next()) {
+                $tableEditor = $tools->GetTableEditorManager($treeTableName, $cmsTreeNode->id);
+                $newSortOrder = $tableEditor->oTableEditor->oTable->sqlData[$entrySortField] + 1;
+                $tableEditor->SaveField('entry_sort', $newSortOrder);
+                $updatedNodes[] = $cmsTreeNode;
+            }
+
+            $tableEditor->Init($tableId, $nodeId);
+            $tableEditor->SaveField('entry_sort', 0);
+            $tableEditor->SaveField('parent_id', $parentNodeId);
+        } else {
+            $quotedNodeId = $this->dbConnection->quote($nodeId);
+            $quotedEntrySortField = $this->dbConnection->quoteIdentifier($entrySortField);
+            $query = 'SELECT * FROM ' . $quotedTreeTable . ' WHERE parent_id = ' . $quotedParentNodeId . ' AND id != ' . $quotedNodeId . ' ORDER BY ' . $quotedEntrySortField . ' ASC';
+            $cmsTreeList = \TdbCmsTreeList::GetList($query, \TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+
+            $count = 0;
+            while ($cmsTree = $cmsTreeList->Next()) {
+                if ($position === $count) { // skip new position of moved node
+                    ++$count;
+                }
+
+                $tableEditor = $tools->GetTableEditorManager($treeTableName, $cmsTree->id);
+                $tableEditor->SaveField('entry_sort', $count);
+                ++$count;
+            }
+
+            $tableEditor->Init($tableId, $nodeId);
+            $tableEditor->SaveField('entry_sort', $position);
+            $tableEditor->SaveField('parent_id', $parentNodeId);
+            $updatedNodes[] = $cmsTree;
+        }
+
+        // update cache
+        // @todo needed? Should be tested
+        // $this->cache->callTrigger($this->treeTable, $nodeId);
+        $this->updateSubtreePathCache($nodeId);
+
+        $node = \TdbCmsTree::GetNewInstance($nodeId);
+        $this->getNestedSetHelper()->updateNode($node);
+        $this->writeSqlLog();
+
+        $updatedNodes[] = $node;
+        $event = new ChangeNavigationTreeNodeEvent($updatedNodes);
+        $this->eventDispatcher->dispatch(CoreEvents::UPDATE_NAVIGATION_TREE_NODE, $event);
+
+        $messageService = $this->getFlashMessageService();
+        if (true === $messageService->consumerHasMessages(\TCMSTableEditorManager::MESSAGE_MANAGER_CONSUMER)) {
+            $messageService->addMessage(\TCMSTableEditorManager::MESSAGE_MANAGER_CONSUMER, 'ERROR_GENERAL');
```

Comment:
We need to look into this. I'm not sure how error handling is implemented here. This looks not right for me. You just added another message here.

Reply:
After a common look, we have decided not to display an error message here.